src/lib/core/TLVBackingStore.h

```diff
@@ -155,6 +155,8 @@ class DLL_EXPORT TLVBackingStore
      *
      */
     virtual CHIP_ERROR FinalizeBuffer(TLVWriter & writer, uint8_t * bufStart, uint32_t bufLen) = 0;
+
+    virtual bool GetNewBufferWillAlwaysFail() { return false; }
```

Comment:
This needs some documentation.  I know the name is self-explanatory, but still worth explaining why one would have a backing store with no "get new buffer" instead of just using a single buffer.